### ISSUE TITLE
Neovim prints control characters on exit

### ISSUE THREAD

**Issue by borisbrodski** [neovim/neovim#17985]
### Neovim version (nvim -v)

NVIM v0.7.0-dev+1385-g6786b6afa

### Vim (not Nvim) behaves the same?

no, any vim

### Operating system/version

Ubuntu 18.04.6

### Terminal name/version

GNOME Terminal, 3.28.2

### $TERM environment variable

xterm-256color

### Installation

appimage

### How to reproduce the issue

$ ./nvim.appimage -u NORC
:q

[>4;0muser@host:~/

### Expected behavior

$ ./nvim.appimage -u NORC
:q

user@host:~/

### Actual behavior

After exiting Neovim (e.g. with `:q`) neovim prints control characters `[>4;0m` on the screen.

**Comment by borisbrodski** [comment]
Same behavior with bash and zsh installed on my Ubuntu 18.04.6.
Same behavior with zsh on the text console without GNOME Terminal (e.g. Ctrl+Alt+F9).

**Comment by bfredl** [comment]
Cannot reproduce. Could you try also with `nvim --clean` ? (NORC might still load some funky code from packages)

**Comment by bfredl** [comment]
Though I tested with gnome-terminal 3.24 VTE 0.66.2. Could that earlier version of gnome-terminal lacks this sequence.

**Comment by borisbrodski** [comment]
> Cannot reproduce. Could you try also with nvim --clean ? (NORC might still load some funky code from packages)

`nvim --clean` produces the same effect.

> Though I tested with gnome-terminal 3.24 VTE 0.66.2. Could that earlier version of gnome-terminal lacks this sequence.

I tested it also in a pure text mode without Xserver and gnome-terminal without by pressing e.g. Ctrl+Alt+F9. Same effect.

**Comment by jamestrew** [comment]
I'm also experiencing this.

```

~/projects/notes master !2 ?5 ❯ nvim -u NONE --clean

[>4;0m%    

```

https://github.com/neovim/neovim/issues/17985

I see the characters are printing _before_ neovim open and not when neovim exits.

 

### Neovim version

NVIM v0.7.0-dev+1395-ge135adcb8

 

### Operating system/version

RHEL7

 

### Terminal name/version

gnome-terminal 3.28.2

 

### Installation

appimage

**Comment by borisbrodski** [comment]
I agree, I managed to see the ´�[>4;0m´ output for a split second **before** neovim appeared. After exiting neovim the `�[>4;0m` message stays as mentioned above.

**Comment by zeertzjq** [comment]
What shell are you using?

**Comment by borisbrodski** [comment]
I tested it with:

* bash
* zsh
* sh without any custom configuration (using `ssh -t username@localhost /bin/sh`)

All of these print `�[>4;0`

**Comment by gpanders** [comment]
>I agree, I managed to see the ´�[>4;0m´ output for a split second before neovim appeared.

This is odd as `CSI > 4 ; 0 m` shouldn't be sent until Neovim exits.

**Comment by zeertzjq** [comment]
This issue can be reproduced when using GNOME Terminal 3.28.2 with VTE 0.52.2, but not with VTE 0.54.0.

**Comment by Shougo** [comment]
> Same behavior with bash and zsh installed on my Ubuntu 18.04.6.

Oh, very old terminal...

**Comment by gpanders** [comment]
> This issue can be reproduced when using GNOME Terminal 3.28.2 with VTE 0.52.2 (which is 3 years and 11 months old), but not with VTE 0.54.0 (which is 3 years and 7 months old).

@zeertzjq What is the value of `VTE_VERSION` for the 0.54.0 VTE terminal?

**Comment by zeertzjq** [comment]
It is `5401`

**Comment by borisbrodski** [comment]
Tested on

* GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)
* zsh 5.4.2 (x86_64-ubuntu-linux-gnu)

Both print `�[>4;0m`

**Comment by gpanders** [comment]
@borisbrodski What terminal emulator?

**Comment by jamestrew** [comment]
I can confirm this issue is resolved on my machine on `NVIM v0.8.0-dev+12-g147cc60d2`
`gnome-terminal 3.28.2`

**Comment by borisbrodski** [comment]
It's resolved with NVIM v0.8.0-dev+52-g89df042a8 :+1: 

Still present in NVIM 0.7.0 though. 

@gpanders 
> What terminal emulator?

GNOME Terminal, 3.28.2
